### ISSUE TITLE
Help output truncated on Node 6.2

### ISSUE THREAD

**Issue by garthk** [remy/nodemon#842]
… because standard output isn't set non-blocking before output and `process.exit`, which in turn inflicts nodejs/node#6456.

Reproduction:

```
$ node --version
v6.2.0
$ nodemon --version
1.9.2
$ nodemon -h
  Usage: nodemon [options] [script.js] [args]

  Options:

  -e, --ext ................ extensions to look for, ie. js,jade,hbs.
  -x, --exec app ........... execute script with "app", ie. -x "python -v".
  -w, --watch dir........... watch directory "dir" or files. use once for
                             each directory or file to watch.
  -i, --ignore ............. ignore specific files or directories.
  -q, --quiet .............. minimise nodemon messages to start/stop only.
  -V, --verbose ............ show detail on what is causing restarts.
  -I, --no-stdin ........... don't try to read from stdin.
  -C, --on-change-only ..... execute script on change only, not startup
  --no-colors .............. disable color output
  -d, --delay n ............ debounce restart for "n" seconds.
  --exitcrash .............. exit on crash, allows use of nodemon with daemon
                             tools like forever.js.
  -v, --version ............ current nodemon version.
  -h, --help ............$
```

Workaround: `nodemon --help | less`, though I have no idea _why_ that helps.

Fix:

```
process.stdout._handle.setBlocking(true);
// output help
process.exit(0)
```

**Comment by isaacs** [comment]
Piping to `less` (or `cat` or `head` or anything) makes it blocking because it makes the stdout a pipe, rather than a tty.  (But also, it's a regression in node v6.)

**Comment by garthk** [comment]
I think we can leave this to Node now. From the [6.2.1 change log](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V6.md#2016-06-02-version-621-current-rvagg):
> **tty:** Default to blocking mode for stdio on OS X. A bug fix in libuv 1.9.0, introduced in Node.js v6.0.0, exposed problems with Node's use of non-blocking stdio, particularly on OS X which has a small output buffer. This change should fix CLI applications that have been having problems with output since Node.js v6.0.0 on OS X. The core team is continuing to address stdio concerns that exist across supported platforms and progress can be tracked at nodejs/node#6980. (Jeremiah Senkpiel) nodejs/node#6895
